libvirt/README.md

```diff
@@ -44,3 +44,222 @@ If you want to set default password for podvm debugging then you can use guestfi
+Core(s) per socket:  6
+Thread(s) per core:  2
+NUMA cell(s):        1
+Memory size:         32600636 KiB
+```
+
+Now you should finally install the Kubernetes operator in the cluster with the help of the [`install_operator.sh`](./install_operator.sh) script. Ensure that you have your IP address exported in the environment, as shown below, then run the install script:
+
+```
+$ cd ../../../libvirt/
+$ export LIBVIRT_IP="192.168.1.107"
+$ export SSH_KEY_FILE="id_rsa"
+$ ./install_operator.sh
+```
+
+If everything goes well you will be able to see the operator's controller manager and cloud-api-adaptor Pods running:
+
+```
+$ kubectl get pods -n confidential-containers-system
+NAME                                              READY   STATUS    RESTARTS   AGE
+cc-operator-controller-manager-5df7584679-5dbmr   2/2     Running   0          3m58s
+cloud-api-adaptor-daemonset-libvirt-vgj2s         1/1     Running   0          3m57s
+$ kubectl logs pod/cloud-api-adaptor-daemonset-libvirt-vgj2s -n confidential-containers-system
++ exec cloud-api-adaptor-libvirt libvirt -uri 'qemu+ssh://wmoschet@192.168.1.107/system?no_verify=1' -data-dir /opt/data-dir -pods-dir /run/peerpod/pods -network-name default -pool-name default -socket /run/peerpod/hypervisor.sock
+2022/11/09 18:18:00 [helper/hypervisor] hypervisor config {/run/peerpod/hypervisor.sock  k8s.gcr.io/pause:3.7 /run/peerpod/pods libvirt}
+2022/11/09 18:18:00 [helper/hypervisor] cloud config {qemu+ssh://wmoschet@192.168.1.107/system?no_verify=1 default default /opt/data-dir}
+2022/11/09 18:18:00 [helper/hypervisor] service config &{qemu+ssh://wmoschet@192.168.1.107/system?no_verify=1 default default /opt/data-dir}
+```
+
+You will also notice that Kubernetes [*runtimeClass*](https://kubernetes.io/docs/concepts/containers/runtime-class/) resources
+were created on the cluster, as for example:
+
+```
+$ kubectl get runtimeclass
+NAME        HANDLER     AGE
+kata        kata        7m41s
+kata-clh    kata-clh    7m41s
+kata-qemu   kata-qemu   7m41s
+```
+
+## Create a sample peer-pods pod
+
+At this point everything should be fine to get a sample Pod created. Let's first list the running VMs so that we can later check
+the Pod VM will be really running. Notice below that we got only the cluster node VMs up:
+
+```
+$ virsh -c qemu:///system list
+ Id   Name                 State
+------------------------------------
+ 3    peer-pods-master-0   running
+ 4    peer-pods-worker-0   running
+```
+
+Create the *sample_busybox.yaml* file with the following content:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    run: busybox
+  name: busybox
+spec:
+  containers:
+  - image: quay.io/prometheus/busybox
+    name: busybox
+    resources: {}
+  dnsPolicy: ClusterFirst
+  restartPolicy: Never
+  runtimeClassName: kata
+```
+
+And create the Pod:
+
+```
+$ kubectl apply -f sample_busybox.yaml
+pod/busybox created
+$ kubectl wait --for=condition=Ready pod/busybox
+pod/busybox condition met
+```
+
```

Comment:
Creating the pod failed for me:
```
Events:
  Type     Reason                  Age   From               Message
  ----     ------                  ----  ----               -------
  Normal   Scheduled               62s   default-scheduler  Successfully assigned default/busybox to peer-pods-worker-0
  Warning  FailedMount             62s   kubelet            MountVolume.SetUp failed for volume "kube-api-access-67gfb" : failed to sync configmap cache: timed out waiting for the condition
  Normal   AddedInterface          57s   multus             Add eth0 [10.244.1.4/24] from cbr0
  Warning  FailedCreatePodSandBox  2s    kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create containerd task: failed to create shim task: remote hypervisor call failed: ttrpc: closed: unknown
```
It's probably user error, but I'm not sure how to resolve it.

Reply:
I got a different error on this set-up:
```
Events:
  Type     Reason                  Age               From               Message
  ----     ------                  ----              ----               -------
  Normal   Scheduled               107s              default-scheduler  Successfully assigned default/busybox to peer-pods-worker-0
  Normal   AddedInterface          102s              multus             Add eth0 [10.244.1.12/24] from cbr0
  Warning  FailedCreatePodSandBox  61s               kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create containerd task: failed to create shim task: remote hypervisor call failed: ttrpc: closed: unknown
  Normal   AddedInterface          53s               multus             Add eth0 [10.244.1.13/24] from cbr0
  Normal   AddedInterface          41s               multus             Add eth0 [10.244.1.14/24] from cbr0
  Normal   AddedInterface          21s               multus             Add eth0 [10.244.1.15/24] from cbr0
  Normal   AddedInterface          12s               multus             Add eth0 [10.244.1.16/24] from cbr0
  Warning  FailedCreatePodSandBox  6s (x4 over 47s)  kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create containerd task: failed to create shim task: failed to connect to remote hypervisor socket: dial unix /run/peerpod/hypervisor.sock: connect: connection refused: unknown
  Normal   AddedInterface          2s                multus             Add eth0 [10.244.1.17/24] from cbr0
```

One thing I'll note is that containerd doesn't seem to be installed, which I'd expect the pre-install section to do  when the operator is installed?
```
# containerd version

Command 'containerd' not found, but can be installed with:

apt install containerd
```

Reply:
On my big boy 8vCPU 32GB RAM the libvirt peer pod worked! (almost). The `kubectl wait --for=condition=Ready pod/busybox` failed as it took just over 30seconds to be created, but I think if we can document that you might need a really big machine then I'm happy with this. Thanks for the patience all!